dotCMS/src/main/java/com/dotcms/rest/SearchForm.java

```diff
@@ -0,0 +1,142 @@
+package com.dotcms.rest;
+
+import com.dotmarketing.business.APILocator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+
+/**
+ * Search Form to make a ES query
+ * @author jsanca
+ */
+@JsonDeserialize(builder = SearchForm.Builder.class)
+public class SearchForm {
+
+    private final String query;
+    private final String sort;
+    private final int limit;
+    private final int offset;
+    private final String userId;
+    private final String render;
+    private final int depth;
+    private final long languageId;
+    private final boolean allCategoriesInfo;
+
+    private SearchForm (final Builder builder) {
+
+        this.query  = builder.query;
+        this.sort   = builder.sort;
+        this.limit  = builder.limit;
+        this.offset = builder.offset;
+        this.userId = builder.userId;
+        this.render = builder.render;
+        this.depth  = builder.depth;
+        this.languageId = builder.languageId;
+        this.allCategoriesInfo = builder.allCategoriesInfo;
+    }
+
+    public String getQuery() {
+        return query;
+    }
+
+    public String getSort() {
+        return sort;
+    }
+
+    public int getLimit() {
+        return limit;
+    }
+
+    public int getOffset() {
+        return offset;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+    public String getRender() {
+        return this.render;
+    }
+
+    public int getDepth() {
+        return depth;
+    }
+
+    public long getLanguageId() {
+        return languageId;
+    }
+
+    public boolean isAllCategoriesInfo() {
+        return allCategoriesInfo;
+    }
+
+    public static final class Builder {
+
+        private  @JsonProperty String query  = "";
+        private  @JsonProperty String sort   = "";
+        private  @JsonProperty int    limit  = 0;
+        private  @JsonProperty int    offset = 20;
+        private  @JsonProperty String userId;
+        private  @JsonProperty String render;
+        private  @JsonProperty int depth       = -1;
```

Comment:
Codacy found an issue: [Field depth has the same name as a method](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6233451)